lib/icedfrisby.js

```diff
@@ -654,11 +654,12 @@ class Frisby {
     header = (header + "").toLowerCase()
     this.current.expects.push(function () {
       if (typeof self.current.response.headers[header] !== "undefined") {
-        chai.expect(self.current.response.headers[header].toLowerCase()).to
-          .contain(content.toLowerCase())
+        var responseHeaders = [].concat(self.current.response.headers[header])
+        chai.expect(responseHeaders).to
+          .include.something.that.satisfies(thisHeader => thisHeader.toLowerCase().includes(content.toLowerCase())) // Ugly workaround for https://github.com/chaijs/chai-things/issues/42
```

Comment:
Is it problematic that we are lowercasing the content of the header?

Reply:
Yes, lowercasing the value is problematic, and also unexpected. We should fix that, and make sure a test is covering it.

I have no problem with lowercasing the field name, since [field names are case-insensitive per the HTTP spec](https://tools.ietf.org/html/rfc7230#section-3.2).

Reply:
You're absolutely right. I went mad lowercasing things after I'd seen one comparison fail because of casing. I'll fix this.

Reply:
Hmmm. The original had the same problem. The `expectHeader` still has the same behaviour. The header's content is lower-cased and compared the input, also lower-cased.

Suggest we separate the concerns and create a separate bug and I'll write a separate PR for this, @paulmelnikow?

Reply:
I've written some unit tests in another branch. Only `expectHeaderToMatch` is failing this right now on master (because we're currently intentionally making a case-insensitive compare everywhere, except in regex)
https://github.com/Fishbowler/IcedFrisby/tree/fix_case_sensitivity

This PR retains this case-insensitive functionality, and the rewritten logic in `expectHeaderToMatch` doesn't have the same problem.

I've merged those tests into this branch. Let me know how you'd like to proceed.

Reply:
Right, the problem existed before. I don't see a problem with fixing it here, while we're already rewriting all that code.

I'm a bit confused. How does the code stand right now? Is it matching case for values?

Reply:
It's currently converting header value and input to lowercase for the comparison. Any danger in breaking tests by users by changing this behaviour?

Reply:
expectHeaderToMatch was failing on master, but is working on this branch. 
The unit tests are written to assume case insensitive matching is intentional. 
In fact, just checked, docs for `expectHeader` and `expectHeaderContains` both explicitly say that the content comparisons are case-insensitive.

Reply:
Gotcha. Will read with that in mind. We can merge it, and make the value matching case-sensitive in a separate PR. Ordinarily I'd consider this a bug fix, but since the behavior was documented I suppose it's a breaking change.